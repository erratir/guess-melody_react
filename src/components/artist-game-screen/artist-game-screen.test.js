import React from "react";
import renderer from "react-test-renderer";
import ArtistGameScreen from "./artist-game-screen";

const mock = {
  type: `artist`,
  song: {
    src: ``,
    artist: `Jon Snow`,
  },
  answers: [
    {
      pic: ``,
      artist: `Mr Putin`,
    },
    {
      pic: ``,
      artist: `Jon Snow`,
    },
    {
      pic: ``,
      artist: `Mr. Bim`,
    },
  ],
};

it(`ArtistGameScreen is rendered correctly`, () => {
  const {song, answers} = mock;
  const tree = renderer.create(<ArtistGameScreen
    onAnswer = {jest.fn()}
    song={song}
    answers={answers}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
