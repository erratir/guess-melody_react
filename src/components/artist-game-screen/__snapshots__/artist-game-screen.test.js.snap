// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistGameScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Mr Putin"
          className="artist__picture"
          src=""
        />
        Mr Putin
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Jon Snow"
          className="artist__picture"
          src=""
        />
        Jon Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Mr. Bim"
          className="artist__picture"
          src=""
        />
        Mr. Bim
      </label>
    </div>
  </form>
</section>
`;
